### DISCUSSION TITLE
How to choose ALSA sound device on Raspberry Pi + Ubuntu?

### DISCUSSION THREAD

**Discussion by surajang** [grame-cncm/faust#935] [Q&A] [answered]
Hello, FAUSTers!

So I made a simple sound generating thing with FAUST and built a Linux CLI app using faust2alsaconsole script.
I ran it on Raspberry Pi 4 with Ubuntu 20.04 installed and it worked....kinda :/ 
The thing is, Raspberry Pi has two audio output options: one is HDMI audio(device 0) and the other is analog 3.5mm headphone jack(device 1).
**I wanted to my app output to headphone jack**, however I have no idea how to do it.
One thing I tried is to set default sound output to headphone jack by editing `.asoundrc` file. It seemed like work however, my FAUST CLI app still output to HDMI audio. Seems like the app always choose the device 0 for its output destination.
Any idea?

Thx

**Comment by sletz** [comment, marked as answer]
You may have to improve the (now quite old) ALSA architecture file here: https://github.com/grame-cncm/faust/blob/master-dev/architecture/faust/audio/alsa-dsp.h

**Reply by surajang** [reply]
Thanks for the nice starting point :)
Found [this line](https://github.com/grame-cncm/faust/blob/a6e1a775f7b8ae542f968716aca7983b1429a620/architecture/faust/audio/alsa-dsp.h#L604) and I could pass `-d` parameter with `"hw:1"` and it works!!
